### PR TITLE
Simplify code paths

### PR DESCRIPTION
**Pull request by kix**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Documented?   | nothing to document
| Fixed tickets | none
| License       | MIT

This simplifies several `if` statements throughout the code

### CODE REVIEW

File: src/CacheWarmer/CompileCacheWarmer.php [hunk 1]
```diff
@@ -38,14 +38,16 @@ public function isOptional()
...
-        if ($this->compiled) {
-            // use warm up cache dir if type generator cache dir not already explicitly declare
-            $baseCacheDir = $this->typeGenerator->getBaseCacheDir();
-            if (null === $this->typeGenerator->getCacheDir(false)) {
-                $this->typeGenerator->setBaseCacheDir($cacheDir);
-            }
-            $this->typeGenerator->compile(TypeGenerator::MODE_WRITE | TypeGenerator::MODE_OVERRIDE);
-            $this->typeGenerator->setBaseCacheDir($baseCacheDir);
+        if (!$this->compiled) {
+          return;
```
Comment (murtukov) [inline src/CacheWarmer/CompileCacheWarmer.php]:
```suggestion
            return;
```

Fix CS: add 2 more spaces

File: src/Definition/ConfigProcessor/AclConfigProcessor.php [hunk 2]
```diff
@@ -73,13 +73,13 @@ public function process(LazyConfig $lazyConfig): LazyConfig
     private static function findFieldResolver(array $field, ResolveInfo $info, callable $defaultResolver): callable
     {
         if (isset($field['resolve'])) {
-            $resolver = $field['resolve'];
-        } elseif (isset($info->parentType->config['resolveField'])) {
-            $resolver = $info->parentType->config['resolveField'];
-        } else {
-            $resolver = $defaultResolver;
+            return $field['resolve'];
+        }
```
Comment (murtukov) [inline src/Definition/ConfigProcessor/AclConfigProcessor.php]:
```suggestion
        }
```

Fix CS: remove trailing whitespace

File: src/Config/Parser/AnnotationParser.php [hunk 3]
```diff
@@ -490,16 +490,17 @@ private static function unionAnnotationToGQLConfiguration(string $className, GQL
...
-                }
-            } else {
-                throw new InvalidArgumentException(\sprintf('The annotation @Union has no "resolveType" attribute and the related class has no "resolveType()" public static method. You need to define of them.'));
+            if (!isset($methods['resolveType'])) {
+              throw new InvalidArgumentException(\sprintf('The annotation @Union has no "resolveType" attribute and the related class has no "resolveType()" public static method. You need to define of them.'));
+            }
+
+            $method = $methods['resolveType']['method'];
+
+            if (!$method->isStatic() ||!$method->isPublic()) {
```
Comment (murtukov) [inline src/Config/Parser/AnnotationParser.php]:
```suggestion
            if (!$method->isStatic() || !$method->isPublic()) {
```

Fix CS: add space around binary operator

File: src/Config/Parser/AnnotationParser.php [hunk 4]
```diff
@@ -871,22 +872,23 @@ private static function guessType(string $namespace, array $annotations): string
...
-                if ($isMultiple) {
-                    return \sprintf('[%s]!', $type);
-                } else {
-                    $isNullable = false;
-                    $joinColumn = self::getFirstAnnotationMatching($annotations, JoinColumn::class);
-                    if ($joinColumn) {
-                        $isNullable = $joinColumn->nullable;
-                    }
+            if (!$type) {
+              throw new \RuntimeException(\sprintf('Unable to auto-guess GraphQL type from Doctrine target class "%s" (check if the target class is a GraphQL type itself (with a @GQL\Type annotation).', $target));
```
Comment (murtukov) [inline src/Config/Parser/AnnotationParser.php]:
```suggestion
                throw new \RuntimeException(\sprintf('Unable to auto-guess GraphQL type from Doctrine target class "%s" (check if the target class is a GraphQL type itself (with a @GQL\Type annotation).', $target));
```

Fix CS: add 2 more spaces before `throw`

File: src/Relay/Connection/Paginator.php [hunk 5]
```diff
@@ -76,16 +76,16 @@ public function forward(ArgumentInterface $args)
...
-            $entities = \call_user_func($this->fetcher, $offset, $limit ? $limit + 2 : $limit);
-
-            return $this->handleEntities($entities, function ($entities) use ($args, $offset) {
-                return $this->connectionBuilder->connectionFromArraySlice($entities, $args, [
-                    'sliceStart' => $offset,
-                    'arrayLength' => $offset + \count($entities),
-                ]);
-            });
         }
+
```
Comment (murtukov) [inline src/Relay/Connection/Paginator.php]:
```suggestion
```
Fix CS: remove whitespaces in a blank line

**Comment by kix** [comment]
@murtukov thanks for spotting! Will fix

**Comment by murtukov** [comment]
@kix You just have to accept suggestions that I prepared, no further actions needed.

**Comment by kix** [comment]
All suggestions accepted, thanks. Guess I shouldn't have been editing PHP code in Atom :)

**Comment by murtukov** [comment]
@kix One last suggestion left. I use PHPStorm, it automatically removes unnecessary whitespaces and aligns code properly

**Comment by kix** [comment]
@murtukov, guess I have my PHPStorm misconfigured in some way. Thanks for the suggestion though!

**Comment by murtukov** [comment]
@kix I mean the last suggestion above, you didn't accept it yet 😄

**Comment by kix** [comment]
Oh, right. Got it, thanks. All accepted and squashed into a single commit

File: src/Config/Parser/AnnotationParser.php [hunk 6]
```diff
@@ -490,16 +490,17 @@ private static function unionAnnotationToGQLConfiguration(string $className, GQL
...
-                $method = $methods['resolveType']['method'];
-                if ($method->isStatic() && $method->isPublic()) {
-                    $unionConfiguration['resolveType'] = self::formatExpression(\sprintf("@=call('%s::%s', [service('overblog_graphql.type_resolver'), value], true)", self::formatNamespaceForExpression($className), 'resolveType'));
-                } else {
-                    throw new InvalidArgumentException(\sprintf('The "resolveType()" method on class must be static and public. Or you must define a "resolveType" attribute on the @Union annotation.'));
-                }
-            } else {
-                throw new InvalidArgumentException(\sprintf('The annotation @Union has no "resolveType" attribute and the related class has no "resolveType()" public static method. You need to define of them.'));
+            if (!isset($methods['resolveType'])) {
+              throw new InvalidArgumentException(\sprintf('The annotation @Union has no "resolveType" attribute and the related class has no "resolveType()" public static method. You need to define of them.'));
```
Comment (murtukov) [inline src/Config/Parser/AnnotationParser.php]:
```suggestion
                throw new InvalidArgumentException(\sprintf('The annotation @Union has no "resolveType" attribute and the related class has no "resolveType()" public static method. You need to define of them.'));
```

Fix CS: add 2 more spaces before `throw`

**Comment by murtukov** [comment]
Hey @kix 

The codebase of the project was heavily upgraded after you opened this PR and it became outdated. Would you like to create this PR again from the new master?

**Comment by kix** [comment]
@murtukov sure, when I have some spare time :)

**Comment by murtukov** [comment]
@kix Thanks. I close this one then.

**Pull request closed without merging**